src/flexmock/_api.py

```diff
@@ -1315,25 +1315,20 @@ def get_flexmock_expectation(
             args = {"kargs": args, "kwargs": {}}
         if not isinstance(args["kargs"], tuple):
             args["kargs"] = (args["kargs"],)
-        if name and obj in cls.flexmock_objects:
-            found = None
-            for expectation in reversed(cls.flexmock_objects[obj]):
-                if expectation._name == name and expectation._match_args(args):
-                    if expectation in cls.ordered or not expectation._ordered and not found:
-                        found = expectation
-            if found and found._ordered:
-                cls._verify_call_order(found, args)
-            return found
-        return None
+        found = None
+        for expectation in reversed(cls.flexmock_objects[obj]):
+            if expectation._name == name and expectation._match_args(args):
+                if expectation in cls.ordered or not expectation._ordered and not found:
+                    found = expectation
+        if found and found._ordered:
+            cls._verify_call_order(found, args)
+        return found
 
     @classmethod
     def _verify_call_order(cls, expectation: Expectation, args: Dict[str, Any]) -> None:
-        if not cls.ordered:
```

Comment:
Why can't it be empty ? I guess it's because we never instantiate directly ?

Reply:
Yeah this method is only called from `get_flexmock_expectation` and that already checks that the expectation is ordered, so it should not be possible for `cls.ordered` to be empty. From the public API, this code is unreachable.